sockeye/inference.py

```diff
@@ -157,12 +157,13 @@ def _get_encoder_module(self) -> Tuple[mx.mod.BucketingModule, int]:
 
         def sym_gen(source_seq_len: int):
             source = mx.sym.Variable(C.SOURCE_NAME)
-            source_factors = [mx.sym.Variable('%s_factor_%d' % (C.SOURCE_NAME,i)) for i in range(self.num_factors)]
-            source_length = utils.compute_lengths(source)
+            source_words = source.split(num_outputs=self.num_factors + 1, axis=2, squeeze_axis=True)[0]
```

Comment:
can we make this splitting optional if there are no factors to save a split operation in the graph?